hfile/reader.go

```diff
@@ -254,3 +265,74 @@ func (r *Reader) GetIterator() *Iterator {
 		return NewIterator(r)
 	}
 }
+
+// TODO(benjy): Write tests for this, once the writer supports writing the FileInfo block.
+func (r *Reader) readFileInfo(data []byte) (err error) {
+	type outOfBounds struct {}
+
+	defer func() {
+		if r := recover(); r != nil {
+			switch r.(type) {
+			case outOfBounds:
+				err = errors.New("Tried to read past end of FileInfo block!")
+			default:
+				panic(r)
+			}
+		}
+	}()
+
+	r.FileInfo.InfoFields = make(map[string]string)
+
+	fileInfoBlockEnd := r.DataIndexOffset
+
+	p := r.FileInfoOffset
+
+	if p == fileInfoBlockEnd {
+		log.Println("[Reader.readFileInfo] No FileInfo block found. Skipping.")
+		return nil
+	}
+
+	// Safely move the read pointer.
+	advance := func(increment int) {
+		p += uint64(increment)
+		if p > fileInfoBlockEnd {
+			panic(new(outOfBounds))
+		}
+	}
+
+	// Grab a variable-length sequence of bytes from the buffer.
+	varLenBytes := func() []byte {
+		// Read the length of the sequence, as a varint.
+		seqLen, s := vintAndLen(data[p:])
+		advance(s)
+		// Read the sequence itself.
+		seq := data[p : p + uint64(seqLen)]
+		advance(seqLen)
+		return seq
+	}
+
+	// Heuristic to figure out how to print the value in a hopefully human-readable way.
+	printableValue := func(buf []byte) string {
```

Comment:
I'd move this out to helpers -- I think there are already a couple "pretty" print things that did that -- and it'll be easier to test it.